zproject/backends.py

```diff
@@ -868,6 +883,61 @@ def sync_custom_profile_fields_from_ldap(
         except SyncUserError as e:
             raise ZulipLDAPError(str(e)) from e
 
+    def sync_groups_from_ldap(self, user_profile: UserProfile, ldap_user: _LDAPUser) -> None:
+        if user_profile.realm.id not in settings.LDAP_GROUP_SYNC_LIST:
+            return  # no groups synced for this realm
+
+        ldap_group_sync_list = set(settings.LDAP_GROUP_SYNC_LIST[user_profile.realm.id])
+
+        try:
+            ldap_logger.debug("found groups for user: %s", str(user_profile))
```

Comment:
`ldap_logger.debug("Syncing groups for user: %s", user_profile.id)` would be a bit better message + logging just the user id (str(user_profile) will log the user's email and we want to avoid logging personal details like that especiialy when not necessary - so id is better)